mobile/library/cc/engine_builder.cc

```diff
@@ -259,26 +259,46 @@ EngineBuilder& EngineBuilder::enforceTrustChainVerification(bool trust_chain_ver
   return *this;
 }
 
-EngineBuilder& EngineBuilder::addRtdsLayer(const std::string& layer_name,
-                                           const int timeout_seconds) {
-  rtds_layer_name_ = layer_name;
-  rtds_timeout_seconds_ = timeout_seconds;
+EngineBuilder& EngineBuilder::setNodeId(std::string node_id) {
+  node_id_ = std::move(node_id);
+  return *this;
+}
+
+EngineBuilder& EngineBuilder::setNodeLocality(std::string region, std::string zone,
+                                              std::string sub_zone) {
+  set_node_locality_ = true;
+  node_locality_region_ = std::move(region);
+  node_locality_zone_ = std::move(zone);
+  node_locality_sub_zone_ = std::move(sub_zone);
```

Comment:
If these are always set together, would it be better to use something like `std::optional<NodeLocality>` ?

Reply:
Added NodeLocality struct